sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -366,6 +379,20 @@ public static boolean isDefaultE2ETimeoutDisabledForNonPointOperations() {
         return DEFAULT_E2E_FOR_NON_POINT_DISABLED_DEFAULT;
     }
 
+    public static boolean isIdValueValidationEnabled() {
+        String valueFromSystemProperty = System.getProperty(PREVENT_INVALID_ID_CHARS);
+        if (valueFromSystemProperty != null && !valueFromSystemProperty.isEmpty()) {
+            return !Boolean.valueOf(valueFromSystemProperty);
+        }
+
+        String valueFromEnvVariable = System.getenv(PREVENT_INVALID_ID_CHARS_VARIABLE);
+        if (valueFromEnvVariable != null && !valueFromEnvVariable.isEmpty()) {
+            return!Boolean.valueOf(valueFromEnvVariable);
+        }
+
+        return !DEFAULT_PREVENT_INVALID_ID_CHARS;
```

Comment:
should the default value be `false`? instead of using` !`?

Reply:
@FabianMeiswinkel - I also think the default should be `false` - otherwise, while debugging this becomes an issue when we see the default value set as `true`. Someone might easily miss this `!defaultValue` :)

Reply:
Sorry - I do not get this? Reading simple conditional statements is something any developer can do?

Reply:
If we change default value we would also need to change meaning/name of the variables - I can do it but really do not see where the problem is?

Reply:
Fixed